rust_src/src/xdisp.rs

```diff
@@ -37,4 +41,83 @@ pub fn trace_redisplay(arg: Option<InteractiveNumericPrefix>) {
     }
 }
 
+fn invisible_prop(propval: LispObject, list: LispCons) -> EmacsInt {
+    for tem in list.iter_cars(LispConsEndChecks::off, LispConsCircularChecks::off) {
+        if propval.eq(tem) {
+            return 1;
+        }
+        if tem.is_cons() && propval.eq(tem.as_cons().unwrap().car()) {
+            if tem.as_cons().unwrap().cdr().is_nil() {
```

Comment:
You might find this to be a bit more readable:
```
let (car, cdr) = tem.into();
if car.eq(propval) {
    return if cdr.is_nil() { 1 } else { 2 }
}
```

Reply:
Oh, that is quite a bit nicer, thank you!